src/factories/inputs/FactoryNumberInput.tsx

```diff
@@ -0,0 +1,43 @@
+import { NumberInput, type NumberInputProps } from '@mantine/core';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+export function FactoryNumberInput(props: NumberInputProps) {
+  // Very complicated yet simple handlers for "shift for +10" behaviour on the resource number input
+  const [inputStep, setInputStep] = useState(1);
+  const keyDownHandler = useCallback(
+    (e: KeyboardEvent) => e.key == 'Shift' && setInputStep(10),
+    [],
+  );
+  const keyUpHandler = useCallback(
```

Comment:
Why use useCallback instead of setting directly inside the useEffect? Seems like a a simple functionality with extra steps